src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -32,6 +32,29 @@ public abstract class CompletionServiceWithProviders : CompletionService, IEqual
 
         private readonly Workspace _workspace;
 
+        /// <summary>
+        /// For a given language, a <see cref="CompletionServiceWithProviders"/> is created once
+        /// for regular editing scenarios. An additional one is created for each Debugger
+        /// IntelliSense session (e.g., each new Watch Window entry). The provider maps are
+        /// initialized when <see cref="ShouldTriggerCompletion(SourceText, int, CompletionTrigger, ImmutableHashSet{string}, OptionSet)"/>
+        /// is called. In debugger scenarios, the completion service specific to that instance
+        /// is used to trigger completion, but the one created for regular editing scenarios is
+        /// used for showing the Description and for Committing. If you do completion in the 
+        /// debugger before ever doing completion in a regular open document, then the completion
+        /// service being used to commit items hasn't been initialized (that is, 
+        /// ShouldTriggerCompletion has never been called). So, when it's time to lookup the
+        /// provider for the committed item, the cache has not been created, so we don't find
+        /// the provider, and a default provider is used which inserts the DisplayText. This is
+        /// a problem for entries like "List&lt;&gt;", where the insertion text and DisplayText
+        /// differ. 
+        /// Refactoring such that the same completion service is used for triggering and 
+        /// descriptions/committing in debugger scenarios is risky, so we're taking a more
```

Comment:
really?   it seems super strange that we can even get into this situatoin in the first place, and it makes me very quesy about how these services are created and then used.  i mean, it basically hopes/prays htat the one service will have no problem and have no expectations when getting data from another service.  As htis bug indicates, that's clearly not a good assumption to make :)

Reply:
can this be answered?  It seems super weird to me that we even got into this situatoin in the first place.  What codepath it working with multiple CompletionServices... adn why not just fix that codepath?  That seems like teh real fix to have here...